src/content/docs/mobile-monitoring/new-relic-mobile-flutter/monitor-your-flutter-application.mdx

```diff
@@ -0,0 +1,431 @@
+* [Manually report errors](#report-errors)
+
+### Track errors and resources [#errors-and-resources]
+
+set up error tracking and resource tracking. Because NewRelic Mobile Start calls WidgetsFlutterBinding.ensureInitialized, if you are not using NewRelic Mobile Start, you need to call this method prior to calling NewrelicMobile.instance.startAgent.
+
+```dart
+if (Platform.isAndroid) {
+    appToken = AppConfig.androidToken;
+} else if (Platform.isIOS) {
+    appToken = AppConfig.iOSToken;
+}
+
+Config config = Config(
+    accessToken: appToken,
+    analyticsEventEnabled: true,
+    networkErrorRequestEnabled: true,
+    networkRequestEnabled: true,
+    crashReportingEnabled: true,
+    interactionTracingEnabled: true,
+    httpResponseBodyCaptureEnabled: true,
+    loggingEnabled: true,
+    webViewInstrumentation: true,
+    printStatementAsEventsEnabled: true,
+    httpInstrumentationEnabled:true);
+
+// NewrelicMobile.instance.start(config, () {
+//   runApp(MyApp());
+// });
+
+runZonedGuarded(() async {
+    WidgetsFlutterBinding.ensureInitialized();
+    FlutterError.onError = NewrelicMobile.onError;
+    await NewrelicMobile.instance.startAgent(config);
+    runApp(MyApp());
+}, (Object error, StackTrace stackTrace) {
+    NewrelicMobile.instance.recordError(error, stackTrace);
+});
+```
+
+AppToken is platform specific. You need to generate the separate token for Android and iOS apps.
+
+## Track screen navigation events
+
+In order to track navigation events you have to add the NewRelicNavigationObserver to your MaterialApp, WidgetsApp or CupertinoApp.
+
+You should provide a name to route settings: RouteSettings(name: 'Your Route Name'). The root route name / will be replaced by root "/" for clarity's sake.
+
+```dart
+import 'package:newrelic_mobile/newrelic_navigation_observer.dart';
+
+//....
+
+MaterialApp(
+    navigatorObservers: [
+        NewRelicNavigationObserver(),
+    ],
+    // other parameters
+)
+```
+
+### GoRouter: Enable automatic routing instrumentation [#screen-nav-events]
+
+When using the [go_router library](https://pub.dev/packages/go_router), the automatic routing instrumentation can be enabled by adding an instance of NewRelicNavigationObserver to your application's GoRouter.observers:
+
+```dart
+//....
+
+import 'package:go_router/go_router.dart';
+import 'package:newrelic_mobile/newrelic_navigation_observer.dart';
+
+
+final router = GoRouter(
+    routes: ...,
+        observers: [NewRelicNavigationObserver()],
+);
+```
+### Manually report errors [#report-errors]
+
+You can manually report non-fatal exceptions using custom attributes in the following method:
+
+```dart
+try {
+    some_code_that_throws_error();
+} catch (ex) {
+NewrelicMobile.instance
+        .recordError(error, StackTrace.current, attributes: attributes);
+}
+```
+
+### Create custom events and metrics
+
+To create custom events and metrics, you'll need to use the iOS or Android SDK APIs.
```

Comment:
@ndesai-newrelic @JustinRush-NR Can you confirm if this is a correct summary of the usage examples?